http/src/main/java/io/micronaut/http/HttpMethod.java

```diff
@@ -100,8 +101,23 @@ public static boolean requiresRequestBody(HttpMethod method) {
      * @return True if it does
      */
     public static boolean permitsRequestBody(HttpMethod method) {
-        return requiresRequestBody(method)
-            || method.equals(OPTIONS)
-            || method.equals(DELETE);
+        return method != null && (requiresRequestBody(method)
+                || method.equals(OPTIONS)
+                || method.equals(DELETE)
+                || method.equals(CUSTOM)
```

Comment:
Why should we assume the method CUSTOM permits a body?

Reply:
Well, I am not sure about that. Some methods definitely do that (for example, REPORT in webdav protocol), so we cannot state otherwise. I can provide СustomHttpMethod with additional property body with values NONE|PERMITS|REQUIRES, but this is definitely demanding a much serious rewrite of the checks. If I drop CUSTOM from this method check, it would still be possible to send body for it, just like for GET, for example?